cmd/juju/cloud/remove_test.go

```diff
@@ -76,57 +76,77 @@ clouds:
+	c.Assert(cmdtesting.Stderr(ctx), gc.Equals, "Removed details of cloud \"homestack\" from the client\n")
 	assertPersonalClouds(c, "homestack2")
 }
 
 func (s *removeSuite) TestRemoveCloudNoControllers(c *gc.C) {
 	s.store.Controllers = nil
-	cmd := cloud.NewRemoveCloudCommandForTest(
+	command := cloud.NewRemoveCloudCommandForTest(
 		s.store,
-		func(controllerName string) (cloud.RemoveCloudAPI, error) {
+		func() (cloud.RemoveCloudAPI, error) {
 			c.Fail()
 			return s.api, nil
 		})
 	s.createTestCloudData(c)
 	assertPersonalClouds(c, "homestack", "homestack2")
-	_, err := cmdtesting.RunCommand(c, cmd, "homestack")
-	c.Assert(err, gc.NotNil)
-	msg := err.Error()
-	msg = strings.Replace(msg, "\n", "", -1)
-	c.Assert(msg, gc.Matches, `There are no controllers running.To remove cloud "homestack" from this client, use the --client-only option.*`)
+	ctx, err := cmdtesting.RunCommand(c, command, "homestack")
+	c.Assert(err, gc.Equals, cmd.ErrSilent)
+	assertPersonalClouds(c, "homestack2")
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, ``)
+	c.Assert(cmdtesting.Stderr(ctx), gc.Matches,
+		"To remove cloud \"homestack\" from this client, use the --client-only option.\n"+
+			"Removed details of cloud \"homestack\" from the client\n"+
+			"Could not remote a cloud from a controller: no controller was specified.\n")
 }
 
-func (s *removeSuite) TestRemoveCloudController(c *gc.C) {
-	var controllerAPICalled string
-	cmd := cloud.NewRemoveCloudCommandForTest(
+func (s *removeSuite) TestRemoveCloudControllerControllerOnly(c *gc.C) {
+	command := cloud.NewRemoveCloudCommandForTest(
 		s.store,
-		func(controllerName string) (cloud.RemoveCloudAPI, error) {
-			controllerAPICalled = controllerName
+		func() (cloud.RemoveCloudAPI, error) {
 			return s.api, nil
 		})
-	ctx, err := cmdtesting.RunCommand(c, cmd, "homestack")
+	s.createTestCloudData(c)
+	ctx, err := cmdtesting.RunCommand(c, command, "homestack", "--no-prompt", "--controller-only")
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(controllerAPICalled, gc.Equals, "mycontroller")
+	assertPersonalClouds(c, "homestack", "homestack2")
+	c.Assert(command.ControllerName, gc.Equals, "mycontroller")
 	s.api.CheckCallNames(c, "RemoveCloud", "Close")
 	c.Assert(cmdtesting.Stderr(ctx), gc.Equals, "Cloud \"homestack\" on controller \"mycontroller\" removed\n")
 }
 
+func (s *removeSuite) TestRemoveCloudController(c *gc.C) {
+	command := cloud.NewRemoveCloudCommandForTest(
+		s.store,
+		func() (cloud.RemoveCloudAPI, error) {
+			return s.api, nil
+		})
+	s.createTestCloudData(c)
+	ctx, err := cmdtesting.RunCommand(c, command, "homestack", "--no-prompt")
+	c.Assert(err, jc.ErrorIsNil)
+	assertPersonalClouds(c, "homestack2")
+	c.Assert(command.ControllerName, gc.Equals, "mycontroller")
+	s.api.CheckCallNames(c, "RemoveCloud", "Close")
+	c.Assert(cmdtesting.Stderr(ctx), gc.Equals,
+		"Removed details of cloud \"homestack\" from the client\n"+
+			"Cloud \"homestack\" on controller \"mycontroller\" removed\n")
```

Comment:
Should the messages be consistent in style?

Either 
Removed details of cloud "mary" from [the client | controller "fred"]
or
Cloud "mary" on [controller "fred" | client] removed

I prefer the former